src/common/ownsql.cpp

```diff
@@ -228,28 +236,29 @@ SqlQuery::~SqlQuery()
     if (_stmt) {
         finish();
     }
+    if (_sqldb) {
+        _sqldb->_queries.remove(this);
```

Comment:
`_sqldb` can be a pointer to a deleted value here. If you use the `SqlQuery(SqlDatabase&)` constructor and never `prepare()` the query can be deleted after the database. I suspect you always want to register queries with the database, not only when `prepare()` succeeds.

Reply:
right. I need to fix this.